docs/source/monitoring.md

```diff
@@ -1,76 +1,148 @@
+* [Deploying Scylla on a Kubernetes Cluster](generic.md)
+* [Deploying Scylla stack using Helm Charts](helm.md)
+
+The above example of the monitoring setup also makes use of HAProxy Ingress and Prometheus Operator.
+You can deploy them in your Kubernetes cluster using the provided third party examples. If you already have them deployed
+in your cluster, you can skip the below steps.
+
+#### Deploy Prometheus Operator
+Deploy Prometheus Operator using kubectl:
+```console
+$ kubectl -n prometheus-operator apply --server-side -f ./examples/third-party/prometheus-operator
+```
+
+##### Verify Prometheus Operator status
+```console
+$ kubectl -n prometheus-operator rollout status --timeout=5m deployments.apps/prometheus-operator
+deployment "prometheus-operator" successfully rolled out
+```
+
+#### Deploy HAProxy Ingress
+Deploy HAProxy Ingress using kubectl:
+```console
+$ kubectl -n haproxy-ingress apply --server-side -f ./examples/third-party/haproxy-ingress
+```
+
+##### Verify HAProxy Ingress status
+```console
+$ kubectl -n haproxy-ingress rollout status --timeout=5m deployments.apps/haproxy-ingress
+deployment "haproxy-ingress" successfully rolled out
+```
+
+Before you begin, update the `endpointSelector` in `examples/monitoring/v1alpha1/scylladbmonitoring.yaml` with a label 
+matching your ScyllaCluster instance name.
+
+### Deploy ScyllaDBMonitoring
+
+Deploy the monitoring setup using kubectl:
+```console
+$ kubectl apply --server-side -f ./examples/monitoring/v1alpha1/scylladbmonitoring.yaml
+```
+
+Scylla Operator will notice the new ScyllaDBMonitoring object, and it will reconcile all necessary resources.
+
+#### Verify ScyllaDBMonitoring status
+```console
+$ kubectl wait --for='condition=Available=True' scylladbmonitorings.scylla.scylladb.com/example && \
+kubectl wait --for='condition=Progressing=False' scylladbmonitorings.scylla.scylladb.com/example && \
+kubectl wait --for='condition=Degraded=False' scylladbmonitorings.scylla.scylladb.com/example
+scylladbmonitoring.scylla.scylladb.com/example condition met
+scylladbmonitoring.scylla.scylladb.com/example condition met
+scylladbmonitoring.scylla.scylladb.com/example condition met
+```
+
+#### Verify Prometheus status
+```console
+$ kubectl rollout status --timeout=5m statefulset.apps/prometheus-example
+statefulset rolling update complete 1 pods at revision prometheus-example-65b89d55bb...
+```
+
+#### Verify Grafana status
+```console
+$ kubectl rollout status --timeout=5m deployments.apps/example-grafana
+deployment "example-grafana" successfully rolled out
+```
+
+#### Verify Grafana admin credentials secret
+Among other resources, the Operator creates a secret containing the admin credentials for you Grafana deployment.
+```console
+$ kubectl get secrets/example-grafana-admin-credentials
+NAME                                TYPE     DATA   AGE
+example-grafana-admin-credentials   Opaque   2      68s
+```
+
+### Access Grafana
```

Comment:
I feel that this article is too focused on how to hack it with minikube and someone trying to set this up on a real cluster / domain would be confused.

I think the generic steps are:
- create ScyllaDBMonitoring with a domain (`test-grafana.test.svc.cluster.local` is fine)
- [real domain / cname to ingress controller] curl
- [fake domain / port-forward / not a cname to ingress controller] curl + `--resolve`

Reply:
> - [real domain / cname to ingress controller] curl
> - [fake domain / port-forward / not a cname to ingress controller] curl + --resolve

I don't understand these lines. Could you elaborate on what these options in the brackets mean? Do you mean that there should be two sections demonstrating all of these options, verifying them with either curl or curl --resolve? Are you also suggesting not using minikube to demonstrate it?

Reply:
I think that could be a joint section saying if your scylldbmonitoring domain isn't resolvable (to the ingress controller IP accessible from outside of the cluster), then you have to add the --resolve flag with ingress controller IP or use `/etc/hosts` alias. The ingress controller IP (accessible from outside of the cluster) can be constructed in a few ways - as service externalIP, nodeIP +  service nodePort, or 127.0.0.1+kubectl port-forward to ingress controller svc, ...